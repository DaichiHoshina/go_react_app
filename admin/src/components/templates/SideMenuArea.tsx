import React, { useEffect, useState } from "react";
import Link from "next/link";
import { List, ListItem, ListItemText } from "@material-ui/core";
import { useRouter } from "next/router";

export type LinkInfo = {
  text: string;
  url: string;
};

const SideMenuArea: React.FC = (): JSX.Element => {
  const [navLinkInfos, setNavLinkInfos] = useState<
    { text: string; url: string }[]
  >([
    { text: "トップページ", url: "/" },
    { text: "ユーザー管理", url: "/users" },
  ]);
  const router = useRouter();

  return (
    <>
      <div className="pt-16">
        <List>
          {navLinkInfos.map((linkInfo) => (
            <Link href={linkInfo.url} key={linkInfo.text}>
              <div
                className={
                  router.pathname.includes(linkInfo.url) ? "bg-gray-300" : ""
                }
                key={linkInfo.text}
              >
                <ListItem button key={linkInfo.url}>
                  <ListItemText primary={linkInfo.text} />
                </ListItem>
              </div>
            </Link>
          ))}
        </List>
      </div>
    </>
  );
};

export default SideMenuArea;
