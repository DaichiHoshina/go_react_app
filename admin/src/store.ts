import { combineReducers } from 'redux';
import logger from 'redux-logger';
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { userSlice, initialState as userInitialState } from './modules/User';

const rootReducer = combineReducers({
  userState: userSlice.reducer,
});

const preloadedState = () => {
  return { userState: userInitialState };
};

const store = () => {
  const middlewareList = [...getDefaultMiddleware(), logger];

  return configureStore({
    reducer: rootReducer,
    middleware: middlewareList,
    devTools: process.env.NODE_ENV !== 'production',
    preloadedState: preloadedState(),
  });
};

export default store;
