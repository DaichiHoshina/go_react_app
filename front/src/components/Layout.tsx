import React, { useEffect, useState } from "react";
import CssBaseline from "@material-ui/core/CssBaseline";
import Box from "@material-ui/core/Box";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Footer from "./organisms/common/Footer";
import { Link } from "@material-ui/core";
import { TUser } from "../modules/User";
import HeaderUserIcon from "../components/atoms/share/HeaderUserIcon";
import { loginConfirm } from "../services/User";
import { useDispatch } from "react-redux";
import LoginIcon from "./atoms/share/LoginIcon";
export interface LayoutProps {
  children?: React.ReactNode;
  title: string;
}

export const Layout: React.FC<LayoutProps> = ({ children, title }) => {
  const dispatch = useDispatch();
  const [isLogin, setIsLogin] = useState(false);

  // ログイン中か確認する
  const userLoginConfirm = async () => {
    const result: any = await dispatch(loginConfirm());
    const user: TUser = result.payload.data;
    setIsLogin(!!user);
  };

  useEffect(() => {
    userLoginConfirm();
  }, []);

  return (
    <div className="min-h-screen pt-10">
      <CssBaseline />
      <AppBar
        position="fixed"
        style={{ zIndex: 1201, color: "#FFFFFF", backgroundColor: "#000000" }}
      >
        <div className="flex">
          <div className="mt-2">
            <Toolbar variant="dense">
              <Link href="/presentations" color="inherit" underline="none">
                {/* タイトルロゴ */}
                <Typography variant="h6" color="inherit" className="flex-grow">
                  Repgram
                </Typography>
              </Link>
            </Toolbar>
          </div>
          {/* 右上のアイコン */}
          <div className="ml-auto mt-2 mr-2">
            {isLogin ? <HeaderUserIcon /> : <LoginIcon />}
          </div>
        </div>
      </AppBar>
      <div className="p-7 mt-3">
        <Typography component="h2" variant="h6" color="inherit">
          {title}
        </Typography>
        <div>{children}</div>
      </div>
      <Box className="pt-8">
        <Footer />
      </Box>
    </div>
  );
};

export default Layout;
