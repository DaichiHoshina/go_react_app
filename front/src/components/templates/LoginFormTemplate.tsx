import { FC } from "react";
import LoginForm from "../organisms/login/LoginForm";

const LoginFormTemplate: FC = () => {
  return (
    <>
      <div className="flex justify-center p-24">
        <LoginForm isSubmit={false} />
      </div>
    </>
  );
};
export default LoginFormTemplate;
