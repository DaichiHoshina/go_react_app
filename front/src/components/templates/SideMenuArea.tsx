import React, { useState } from "react";
import Link from "next/link";
import { List, ListItem, ListItemText } from "@material-ui/core";

export type LinkInfo = {
  text: string;
  url: string;
};

const SideMenuArea: React.FC = (): JSX.Element => {
  const [navLinkInfos] = useState<{ text: string; url: string }[]>([
    { text: "トップページ", url: "/" },
    { text: "ユーザー管理", url: "/users" },
  ]);

  return (
    <div className="pt-16">
      <List>
        {navLinkInfos.map((linkInfo) => (
          <Link href={linkInfo.url} key={linkInfo.text}>
            <div key={linkInfo.text}>
              <ListItem button key={linkInfo.url}>
                <ListItemText primary={linkInfo.text} />
              </ListItem>
            </div>
          </Link>
        ))}
      </List>
    </div>
  );
};

export default SideMenuArea;
