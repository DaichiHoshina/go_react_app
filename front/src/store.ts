import { combineReducers } from "redux";
import logger from "redux-logger";
import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import { userSlice, initialState as userInitialState } from "./modules/User";
import {
  presentationSlice,
  initialState as presentationInitialState,
} from "./modules/Presentation";

const rootReducer = combineReducers({
  userState: userSlice.reducer,
  presentationState: presentationSlice.reducer,
});

const preloadedState = () => {
  return {
    userState: userInitialState,
    presentationState: presentationInitialState,
  };
};

const store = () => {
  const middlewareList = [...getDefaultMiddleware(), logger];

  return configureStore({
    reducer: rootReducer,
    middleware: middlewareList,
    devTools: process.env.NODE_ENV !== "production",
    preloadedState: preloadedState(),
  });
};

export default store;
