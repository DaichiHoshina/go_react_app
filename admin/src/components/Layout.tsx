import React, { useEffect, useRef, useState } from "react";
import CssBaseline from "@material-ui/core/CssBaseline";
import Drawer from "@material-ui/core/Drawer";
import Box from "@material-ui/core/Box";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import MenuIcon from "@material-ui/icons/Menu";
import IconButton from "@material-ui/core/IconButton";
import { useRouter } from "next/router";
import SideMenuArea from "./templates/SideMenuArea";

const drawerWidth = 240;

const Copyright = () => {
  return (
    <Typography variant="body2" color="textSecondary" align="center">
      {"Copyright © Daichi Hoshina"}
    </Typography>
  );
};

export interface LayoutProps {
  children: React.ReactNode;
  title: string;
}

const Layout: React.FC<LayoutProps> = ({ children, title }) => {
  const [isOpen, setIsOpen] = useState(true);
  const sideMenuRef = useRef<HTMLDivElement>(null);
  const router = useRouter();
  const [sideMenuWidth, setSideMenuWidth] = useState(0);

  const handleDrawerOpen = () => {
    setIsOpen(!isOpen);
  };

  useEffect(() => {
    if (sideMenuRef.current !== null) {
      setSideMenuWidth(sideMenuRef.current.clientWidth);
    }
  });

  return (
    <div className="min-h-screen bg-blue-50 pt-10">
      <CssBaseline />
      <AppBar style={{ zIndex: 1201 }} position="fixed">
        <Toolbar variant="dense">
          <IconButton
            edge="start"
            color="inherit"
            aria-label="menu"
            onClick={handleDrawerOpen}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" color="inherit" className="flex-grow">
            管理画面
          </Typography>
        </Toolbar>
      </AppBar>
      <div
        style={{
          paddingLeft:
            isOpen && !router.pathname.includes("/login") ? sideMenuWidth : 0,
        }}
      >
        <div className="p-7">
          <Typography component="h2" variant="h6" color="inherit">
            {title}
          </Typography>
          {children}
        </div>
      </div>
      <Box className="pt-8">
        <Copyright />
      </Box>
      <Drawer open={isOpen} variant="persistent" anchor="left">
        <div ref={sideMenuRef} />
        <SideMenuArea />
      </Drawer>
    </div>
  );
};

export default Layout;
