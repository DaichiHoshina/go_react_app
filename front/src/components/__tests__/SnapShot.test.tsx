import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../../store";
import Layout from "../Layout";
import LoginFormTemplate from "../templates/LoginFormTemplate";
import SideMenuArea from "../templates/SideMenuArea";

it("<Layout>のスナップショット", () => {
  const tree = renderer
    .create(
      <Provider store={store()}>
        <Layout title="" children="" />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("<LoginFormTemplete>のスナップショット", () => {
  const tree = renderer
    .create(
      <Provider store={store()}>
        <LoginFormTemplate />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("<SideMenuArea>のスナップショット", () => {
  const tree = renderer
    .create(
      <Provider store={store()}>
        <SideMenuArea />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
