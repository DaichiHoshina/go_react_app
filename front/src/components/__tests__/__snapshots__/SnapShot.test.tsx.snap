// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout>のスナップショット 1`] = `
<div
  className="min-h-screen pt-10"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    style={
      Object {
        "backgroundColor": "#000000",
        "color": "#FFFFFF",
        "zIndex": 1201,
      }
    }
  >
    <div
      className="flex"
    >
      <div
        className="mt-2"
      >
        <div
          className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineNone MuiTypography-colorInherit"
            href="/presentations"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            <h6
              className="MuiTypography-root flex-grow MuiTypography-h6 MuiTypography-colorInherit"
            >
              Repgram
            </h6>
          </a>
        </div>
      </div>
      <div
        className="ml-auto mt-2 mr-2"
      >
        <div
          className="flex"
        >
          <div
            className="mr-2"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Sign In
              </span>
            </button>
          </div>
          <div>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Sign Up
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </header>
  <div
    className="p-7 mt-3"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
    >
      
    </h2>
    
  </div>
  <div
    className="MuiBox-root MuiBox-root-1 pt-8"
  >
    <div
      className="mt-5 pt-5 text-center text-xs border-t border-gray-300"
    >
      Copyright 2021 Daichi Hoshina
    </div>
  </div>
</div>
`;

exports[`<LoginFormTemplete>のスナップショット 1`] = `
<div
  className="flex justify-center p-24"
>
  <div
    className="MuiPaper-root MuiCard-root flex-item flex flex-col px-8 py-12 w-1/2 max-w-5xl MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "backgroundColor": "#242323",
        "color": "#ffffff",
      }
    }
  >
    <h1
      className="pb-8 text-center text-xl"
    >
      Sign In
    </h1>
    <form
      action=""
      onSubmit={[Function]}
    >
      <ul
        className="pb-10 space-y-3"
      >
        <li
          className=""
        >
          <p
            className="w-48"
          >
            email
            <span
              className="mx-3"
            >
              :
            </span>
          </p>
          <div
            className="bg-gray-700"
          >
            <div
              className="w-full"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="outlined-basic"
                    name="email"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder=""
                    required={false}
                    rows={1}
                    style={
                      Object {
                        "padding": 8,
                        "paddingLeft": 10,
                        "paddingRight": 10,
                      }
                    }
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-3"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          className=""
        >
          <p
            className="w-48"
          >
            password
            <span
              className="mx-3"
            >
              :
            </span>
          </p>
          <div
            className="bg-gray-700"
          >
            <div
              className="w-full"
            >
              <div
                className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              >
                
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    disabled={false}
                    id="outlined-basic"
                    name="password"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder=""
                    required={false}
                    rows={1}
                    style={
                      Object {
                        "padding": 8,
                        "paddingLeft": 10,
                        "paddingRight": 10,
                      }
                    }
                    type="password"
                    value=""
                  />
                  <div
                    className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      className="MuiButtonBase-root MuiIconButton-root"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                          />
                        </svg>
                      </span>
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-3"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained border-none ring-transparent MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <div
        className="mt-5"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained border-none ring-transparent MuiButton-containedSecondary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Login Test User!!
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        ※ポートフォリオ閲覧用です
      </div>
    </form>
  </div>
</div>
`;

exports[`<SideMenuArea>のスナップショット 1`] = `
<div
  className="pt-16"
>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <div
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            トップページ
          </span>
        </div>
      </div>
    </div>
    <div
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            ユーザー管理
          </span>
        </div>
      </div>
    </div>
  </ul>
</div>
`;
